python/lsst/faro/measurement/TractTableMeasurementTasks.py

```diff
@@ -21,14 +21,66 @@
 
 from lsst.pex.config import Config, Field
 from lsst.pipe.base import Struct, Task
+from lsst.pipe.tasks.configurableActions import ConfigurableActionStructField
 from lsst.verify import Measurement, Datum
 
+import lsst.faro.utils.selectors as selectors
 from lsst.faro.utils.tex_table import calculateTEx
 
 import astropy.units as u
 import numpy as np
 
-__all__ = ("TExTableConfig", "TExTableTask")
+__all__ = ("TExTableConfig", "TExTableTask", "NumSourcesConfig", "NumSourcesTask")
+
+
+class NumSourcesConfig(Config):
+    doPrimary = Field(
+        doc="Only count sources where detect_isPrimary is True.",
+        dtype=bool,
+        default=False,
+    )
+
+    selectorActions = ConfigurableActionStructField(
+        doc="Which selectors to use to narrow down the data (independent of band).",
+        default={},
+        # default={"sourceSelector": selectors.StarIdentifier},
+    )
+
+
+class NumSourcesTask(Task):
```

Comment:
There is also a `NumSourcesTask` class in lsst/faro/base/BaseSubTasks. Is it intended that there be two copies of this task?

Reply:
I could use some guidance here. 
If `NumSourcesTask` is imported from lsst/faro/base/BaseSubTasks and configured to contain selectors the code fails with the error pasted below. Either importing the same code from lsst/faro/measurement/TractTableMeasurementTasks or configuring to not use selectors allows the code to run. 

I dont know think we want two versions of NumSources but I dont know which one we should keep 

>lsst.daf.butler.cli.utils ERROR: Caught an exception, details are in traceback:
Traceback (most recent call last):
  File "/software/lsstsw/stack_20210813/stack/miniconda3-py38_4.9.2-0.7.0/Linux64/ctrl_mpexec/g80d878e12a+6dcb4f7875/python/lsst/ctrl/mpexec/cli/cmd/commands.py", line 104, in run
    qgraph = script.qgraph(pipelineObj=pipeline, **kwargs)
  File "/software/lsstsw/stack_20210813/stack/miniconda3-py38_4.9.2-0.7.0/Linux64/ctrl_mpexec/g80d878e12a+6dcb4f7875/python/lsst/ctrl/mpexec/cli/script/qgraph.py", line 183, in qgraph
    qgraph = f.makeGraph(pipelineObj, args)
  File "/software/lsstsw/stack_20210813/stack/miniconda3-py38_4.9.2-0.7.0/Linux64/ctrl_mpexec/g80d878e12a+6dcb4f7875/python/lsst/ctrl/mpexec/cmdLineFwk.py", line 596, in makeGraph
    qgraph = graphBuilder.makeGraph(
  File "/software/lsstsw/stack_20210813/stack/miniconda3-py38_4.9.2-0.7.0/Linux64/pipe_base/g205747afb7+4e808fb5ed/python/lsst/pipe/base/graphBuilder.py", line 1081, in makeGraph
    scaffolding = _PipelineScaffolding(pipeline, registry=self.registry)
  File "/software/lsstsw/stack_20210813/stack/miniconda3-py38_4.9.2-0.7.0/Linux64/pipe_base/g205747afb7+4e808fb5ed/python/lsst/pipe/base/graphBuilder.py", line 455, in __init__
    datasetTypes = PipelineDatasetTypes.fromPipeline(pipeline, registry=registry)
  File "/software/lsstsw/stack_20210813/stack/miniconda3-py38_4.9.2-0.7.0/Linux64/pipe_base/g205747afb7+4e808fb5ed/python/lsst/pipe/base/pipeline.py", line 1052, in fromPipeline
    pipeline = list(pipeline)
  File "/software/lsstsw/stack_20210813/stack/miniconda3-py38_4.9.2-0.7.0/Linux64/pipe_base/g205747afb7+4e808fb5ed/python/lsst/pipe/base/pipeline.py", line 620, in toExpandedPipeline
    yield from self._toExpandedPipelineImpl()
  File "/software/lsstsw/stack_20210813/stack/miniconda3-py38_4.9.2-0.7.0/Linux64/pipe_base/g205747afb7+4e808fb5ed/python/lsst/pipe/base/pipeline.py", line 625, in _toExpandedPipelineImpl
    taskDefs.append(self._buildTaskDef(label))
  File "/software/lsstsw/stack_20210813/stack/miniconda3-py38_4.9.2-0.7.0/Linux64/pipe_base/g205747afb7+4e808fb5ed/python/lsst/pipe/base/pipeline.py", line 668, in _buildTaskDef
    overrides.applyTo(config)
  File "/software/lsstsw/stack_20210813/stack/miniconda3-py38_4.9.2-0.7.0/Linux64/pipe_base/g205747afb7+4e808fb5ed/python/lsst/pipe/base/configOverrides.py", line 291, in applyTo
    exec(override, None, vars)
  File "<string>", line 5, in <module>
  File "/software/lsstsw/stack_20210813/stack/miniconda3-py38_4.9.2-0.7.0/Linux64/pipe_tasks/gec3f545711+8f8c708f41/python/lsst/pipe/tasks/configurableActions/_configurableActionStructField.py", line 160, in __setattr__
    if hasattr(self._config, '_frozen') and self._config._frozen:
  File "/software/lsstsw/stack_20210813/stack/miniconda3-py38_4.9.2-0.7.0/Linux64/pipe_tasks/gec3f545711+8f8c708f41/python/lsst/pipe/tasks/configurableActions/_configurableActionStructField.py", line 146, in _config
    assert(self._config_() is not None)
AssertionError

Reply:
Decided to keep `NumSourcesTask` without selectors for this ticket.